controllers/check/check_iam.go

```diff
@@ -120,56 +118,104 @@ func getDeploymentStatus(r client.Reader, name string) string {
 	return "Ready"
 }
 
-func createUpdateConfigmap(r client.Reader, c client.Client, status string) error {
+func createUpdateConfigmap(bs *bootstrap.Bootstrap, status string) error {
 	cm := &corev1.ConfigMap{}
 	cmName := "ibm-common-services-status"
 	cmNs := "kube-public"
 	if status == "NotReady" {
 		klog.Info("IAM status is NotReady, waiting some minutes...")
 	}
-	err := r.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: cmNs}, cm)
+
+	requestNsSlice := util.GetRequestNs(bs.Reader)
+	err := bs.Reader.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: cmNs}, cm)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			cm.Name = cmName
 			cm.Namespace = cmNs
 			cm.Data = make(map[string]string)
+			for _, requestNs := range requestNsSlice {
+				statusKey := requestNs + "-iamstatus"
+				cm.Data[statusKey] = status
+			}
 			cm.Data["iamstatus"] = status
-			if err := c.Create(context.TODO(), cm); err != nil {
+			if err := bs.Client.Create(context.TODO(), cm); err != nil {
 				klog.Errorf("Failed to create ConfigMap %s: %v", cmName, err)
 				return err
 			}
 			return nil
 		}
 		return err
 	}
-	if cm.Data["iamstatus"] != status {
-		klog.Infof("IAM status is %s", status)
+	isUpdate := false
+	for _, requestNs := range requestNsSlice {
+		statusKey := requestNs + "-iamstatus"
+		// check the iamstatus of cloud pak
+		if val, ok := cm.Data[statusKey]; ok {
```

Comment:
What if ok is false? I believe we need to insert `requestNs + "-iamstatus"` into the configmap anyway.

Reply:
Nice catch!

I was thinking only insert the status when we create the this status Configmap.

But I ignored that the user will update the cs-mapping ConfigMap after the installation and it will introduce new namespaced cloud pak prefix `iamstatus`.

Thank you! It is updated.